mlflow/store/sqlalchemy_store.py

```diff
@@ -68,6 +91,28 @@ def __init__(self, db_uri, default_artifact_root):
             with self.ManagedSessionMaker() as session:
                 self._create_default_experiment(session)
 
+    @staticmethod
+    def _initialize_tables(engine):
+        _logger.info("Creating initial MLflow database tables...")
+        LegacyBase.metadata.create_all(engine)
+        _upgrade_db(str(engine.url))
+
+    @staticmethod
+    def _verify_schema(engine):
+        with engine.connect() as connection:
+            mc = MigrationContext.configure(connection)
+            diff = compare_metadata(mc, Base.metadata)
+            if len(diff) > 0:
+                _logger.error("Detected one or more differences between current database schema "
+                              "and desired schema, exiting. Diff:\n %s",
+                              pprint.pformat(diff, indent=2, width=20))
+                raise MlflowException(
+                    "Detected out-of-date database schema. Take a backup of your database, then "
+                    "run 'mlflow db upgrade %s' to migrate your database to the latest schema. "
```

Comment:
What's `db upgrade`?

Reply:
As per a comment from @dbczumar ([link](https://github.com/mlflow/mlflow/pull/1155#discussion_r279959183)), it seems reasonable to namespace the upgrade command under `mlflow db` to allow flexibility for adding other subcommands, although it does make the command somewhat less discoverable (requires an extra CLI invocation with --help to discover `mlflow db upgrade`, assuming you start with `mlflow --help`)